x-pack/plugins/observability_solution/slo/server/services/create_slo.ts

```diff
@@ -47,62 +48,55 @@ export class CreateSLO {
 
     const rollbackOperations = [];
 
-    await this.repository.save(slo, { throwOnConflict: true });
-    rollbackOperations.push(() => this.repository.deleteById(slo.id));
+    const existingSavedObjectId = await this.repository.getExistingSavedObjectId(slo, {
+      throwOnConflict: true,
+    });
+
+    // can be done async as we are not using the result
+    void this.repository.save(slo, { existingSavedObjectId });
```

Comment:
What happens if the save operation fails when we are already done with the others, and out of the try-catch? I think in general we should not "fire-and-forget" promises.

Also, since we are throwing on conflict, it means existingSavedObjectId will always be undefined, so really here, we are not really interested in getting the internal saved object id but only asserting the non existence of the SLO id.

On top of my other comment regarding the repository change, I would prefer if we were to keep the saving as it was.